### PR TITLE
Rsh/pipeline stages

### PR DESCRIPTION
**Pull request by ramsessanchez**
Fixes #

### Changes proposed in this pull request
- Pipeline is now inline with the Core library structure. 
- Includes Maven Release, Maven-Preview, Github Release stages. 
- Properties_Path and Drop_Path now used as variables to replace repeated file paths. 

### Other links

-

### CODE REVIEW

File: .azure-pipelines/buildAndPackage.yml [hunk 1]
```diff
@@ -22,15 +20,58 @@ trigger:
...
+  - job: Maven-Preview
+    steps: 
+    - template: templates/download-artifacts.yml
+    - template: templates/maven-preview/secure-files.yml
+    - template: templates/maven-preview/copy-and-build.yml
+    - template: templates/alert-failure.yml
 
+- stage: Maven-Release-And-Github-Release
+  dependsOn: []
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```
Comment (baywet) [inline .azure-pipelines/buildAndPackage.yml:53]:
```suggestion
  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
```
for this repo, we don't have a master branch, switch to dev

File: .azure-pipelines/templates/maven-preview/copy-and-build.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    SourceFolder: '$(Agent.TempDirectory)'
+    Contents: '**'
+    TargetFolder: '$(DROP_PATH)/'
+
+- task: Gradle@2
+  inputs:
+    gradleWrapperFile: '$(DROP_PATH)/gradlew'
+    workingDirectory: '$(DROP_PATH)/'
+    options: '-x jar -x javadoc -x javadocJar -x compileJava -x processResources -x classes -x sourceJar'
+    tasks: 'publishSnapshotPublicationToSonatypeSnapshotRepository'
```
Comment (baywet) [inline .azure-pipelines/templates/maven-preview/copy-and-build.yml:13]: set this as a template parameter and merge it with the other copy and build file

File: .azure-pipelines/templates/maven-release/copy-and-build.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+steps:
```
Comment (baywet) [inline .azure-pipelines/templates/maven-release/copy-and-build.yml:1]: remove thanks to template parameter in the other one

File: .azure-pipelines/templates/maven-preview/secure-files.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+steps:
```
Comment (baywet) [inline .azure-pipelines/templates/maven-preview/secure-files.yml:1]: remove thanks to template parameter in the other one

File: .azure-pipelines/templates/maven-release/secure-files.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+steps:
```
Comment (baywet) [inline .azure-pipelines/templates/maven-release/secure-files.yml:1]: add a bolean template parameter "download sign-in files" and add conditions on the last two tasks so we can have a single template.

**Review by baywet — CHANGES_REQUESTED** [review]

File: .azure-pipelines/templates/secure-files.yml [hunk 6]
```diff
@@ -1,9 +1,15 @@
...
+- name: stageID
+  type: string
+  default: ''
+
 steps:
-  - task: DownloadSecureFile@1
-    inputs: 
-      secureFile: 'local.properties'
-      retryCount: '5'
+- task: DownloadSecureFile@1
```
Comment (baywet) [inline .azure-pipelines/templates/secure-files.yml:7]: I think there's a problem with the indent here

Comment (ramsessanchez) [reply .azure-pipelines/templates/secure-files.yml:7]: I believe the indents are correct here. The first task gets called regardless of which stage calls it, so it falls directly under the 'steps' line.

**Review by baywet — APPROVED** [review]

**Pull request merged**